src/pruner.test.ts

```diff
@@ -560,7 +560,34 @@ describe('given array items is object and has nested prop equal to default', ()
 
     const result = pruneDefaults(parameters, schema)
 
+    const expected = {}
+    expect(result).toEqual(expected)
+  })
+})
+
+describe('given an array of booleans where the default is false', () => {
+  it('should replace false with undefined and remove trailing items which are false', () => {
```

Comment:
Good to have this behavior explicitly described in a test, since it's not obvious.